server/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -1972,7 +1986,9 @@ private void innerOpenEngineAndTranslog(LongSupplier globalCheckpointSupplier) t
         // which settings changes could possibly have happened, so here we forcefully push any config changes to the new engine.
         onSettingsChanged();
         assert assertSequenceNumbersInCommit();
-        recoveryState.validateCurrentStage(RecoveryState.Stage.TRANSLOG);
+        recoveryState.validateCurrentStage(
```

Comment:
... I think I'd prefer to do this in `TRANSLOG` stage even for unpromotable shards.